src/hw1/task2.py

```diff
@@ -0,0 +1,49 @@
+import os.path
+
+
+def check_exists(file_path: str):
+    if not os.path.exists(file_path):
+        raise ValueError(f"path to non-existing file: {file_path}")
+
+
+def wc(file_path: str) -> str:
+    check_exists(file_path)
+    line_count = word_count = byte_count = 0
+    with open(file_path) as f:
+        for line in f:
+            line_count += 1
+            word_count += len(line.split(" "))
+            byte_count += len(line)
+
+    return f"{line_count} {word_count} {byte_count} {file_path}"
+
+
+def nl(file_path: str) -> str:
+    check_exists(file_path)
+    newlines = []
+    line_number = 1
+    with open(file_path) as f:
+        for line in f:
+            newlines.append(f"{line_number} {line}")
+            line_number += 1
+    return "".join(newlines)
```

Comment:
If the summing is done around an "empty" space, why not just append to some string all the time?

Reply:
not sure, but appending one by one seems to be more expensive, than when constructing string once from all pieces